vcd/resource_vcd_catalog_org_user_test.go

```diff
@@ -106,24 +112,37 @@ func TestAccVcdCatalogOrgUser(t *testing.T) {
 					resource.TestCheckResourceAttrPair(resourceMedia1, "catalog_id", resourceCatalogOrg1, "id"),
 				),
 			},
+			{
```

Comment:
This one works, but to make the test more effective, we should do the update *after the VM creation*, and check that the VMs are not affected.

Reply:
I made the change, but I have a question regarding it, should there be another step to check if the resources that use the updated name work properly or does Terraform do that automatically?

Reply:
I didn't add the code for checking the VM ID's, but I thought of doing it using the `testCachedFieldValue` struct? You said you have an idea of how to do it, so maybe there's a better way.

Reply:
Using `cacheTestResourceFieldValue` and `testCheckCachedResourceFieldValue` is the right approach.
However, in addition to making sure that the catalog ID has not changed, we need to check that the VM IDs have not changed, i.e. that they were not destroyed and created again.
And since we are at it, let's also check that the ID of the vApp template and catalog media are unchanged.

Reply:
Added the checks. I'm not sure if any other values should be checked as it might be a bit excessive.